internal/controller/accessrequest_controller.go

```diff
@@ -110,12 +108,34 @@ func (r *AccessRequestReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, nil
 	}
 
+	ttlExceeded, err := r.handleTTL(ctx, ar)
+	if err != nil {
+		logger.Error(err, "Error handling TTL")
+		return ctrl.Result{}, fmt.Errorf("error handling TTL: %w", err)
+	}
+	// stop the reconciliation if the TTL was exceeded
+	if ttlExceeded {
+		logger.Debug("AccessRequest TTL exceeded: cleaning up...")
+		return ctrl.Result{}, nil
+	}
+
 	// stop if the reconciliation was previously concluded
 	if isConcluded(ar) {
 		logger.Debug(fmt.Sprintf("Reconciliation concluded as the AccessRequest is %s: skipping...", string(ar.Status.RequestState)))
 		return ctrl.Result{}, nil
 	}
 
+	timeout, err := r.handleTimeout(ctx, ar)
```

Comment:
To be nicer, we should actually validate the timeout at the end of the reconcile loop so we check one last time that the AR does not transition from requested to another state at the deadline.

Reply:
makes sense